src/main/scala/no/ndla/frontpageapi/service/WriteService.scala

```diff
@@ -19,26 +19,51 @@ trait WriteService {
 
   class WriteService {
 
-    def newSubjectPage(subject: api.NewOrUpdateSubjectFrontPageData): Try[api.SubjectPageData] = {
+    def newSubjectPage(subject: api.NewSubjectFrontPageData): Try[api.SubjectPageData] = {
       for {
         convertedSubject <- ConverterService.toDomainSubjectPage(subject)
         subjectPage <- subjectPageRepository.newSubjectPage(convertedSubject, subject.externalId)
-      } yield ConverterService.toApiSubjectPage(subjectPage, "nb")
+        converted <- ConverterService.toApiSubjectPage(subjectPage, "nb")
+      } yield converted
     }
 
-    def updateSubjectPage(id: Long, subject: api.NewOrUpdateSubjectFrontPageData): Try[api.SubjectPageData] = {
+    def updateSubjectPage(id: Long, subject: api.NewSubjectFrontPageData): Try[api.SubjectPageData] = {
       subjectPageRepository.exists(id) match {
         case Success(exists) if exists =>
           for {
             domainSubject <- ConverterService.toDomainSubjectPage(id, subject)
             subjectPage <- subjectPageRepository.updateSubjectPage(domainSubject)
-          } yield ConverterService.toApiSubjectPage(subjectPage, "nb")
+            converted <- ConverterService.toApiSubjectPage(subjectPage, "nb")
+          } yield converted
         case Success(_) =>
           Failure(NotFoundException(id))
         case Failure(ex) => Failure(ex)
       }
     }
 
+    def updateSubjectPage(id: Long, subject: api.UpdatedSubjectFrontPageData): Try[api.SubjectPageData] = {
+      subjectPageRepository.withId(id) match {
+        case Some(existingSubject) =>
+          for {
+            domainSubject <- ConverterService.toDomainSubjectPage(existingSubject, subject)
+            subjectPage <- subjectPageRepository.updateSubjectPage(domainSubject)
+            converted <- ConverterService.toApiSubjectPage(subjectPage, subject.about.get.head.language)
```

Comment:
Språket må nesten finnes ut på en annen måte enn `subject.about.get.head.language`. Både `.get` og `.head` kan kaste exceptions avhengig av hva som sendes inn av brukeren. Enten så må vi tvinge brukeren til å sende inn noe, eller så må vi gjøre som i `updateSubjectPage`funksjonen for `api.NewSubjectFrontPageData` og bare defaulte til "nb" uansett.

Reply:
Endret til at det tas inn fra pathen i patch-requesten, det funker vel bedre da?